rust-engine/src/backends/rust.rs

```diff
@@ -3,19 +3,32 @@
 
 use super::prelude::*;
 
-/// The Rust backend and printer.
-pub struct Rust;
+/// The Rust printer.
+#[derive(Default)]
+pub struct RustPrinter;
+impl_doc_allocator_for!(RustPrinter);
 
-impl Printer for Rust {
+impl Printer for RustPrinter {
     fn resugaring_phases() -> Vec<Box<dyn Resugaring>> {
         vec![]
     }
 
-    const NAME: &str = "Rust";
+    const NAME: &'static str = "Rust";
 }
 
 const INDENT: isize = 4;
 
+/// The Rust backend.
+pub struct RustBackend;
+
+impl Backend for RustBackend {
+    type Printer = RustPrinter;
+
+    fn module_path(&self, _module: &Module) -> std::path::PathBuf {
+        std::path::PathBuf::from("test.rs")
```

Comment:
This should be written into `test.rs`? That seems a little weird. But some documentation around where the output is going and how it's organised should clarify or change this.